lib/adm.js

```diff
@@ -6340,6 +6340,53 @@ maAdm.prototype.createTopology = function createTopology(opts, callback) {
         next();
     }
 
+    /*
+     * This function validates the pnodeToVnodeMap and vnodeToPnodeMap data
+     * structures to ensure the expected number of vnodes are represented in
+     * each structure.
+     */
+    function validateRingVnodeCount(ctx, next) {
+        var mappedVnodeCount = 0;
+        jsprim.forEachKey(
+            ctx.serializedRing.pnodeToVnodeMap,
+            function iterPnode(pnode) {
+                mappedVnodeCount += Object.keys(
+                    ctx.serializedRing.pnodeToVnodeMap[pnode]
+                ).length;
+            }
+        );
+
+        var err;
+        if (mappedVnodeCount !== vnodes) {
+            err = new VError(
+                'The generated hash ring was missing vnode information in the ' +
+                    'pnodeToVnodeMap data structure. Expected vnode count: %d ' +
+                    'Actual vnode count: %d"',
```

Comment:
It looks like there's a rogue `"` at the end of this message. Same as in the vnodeToPnodeMap message.

Reply:
Hah, good eyes! I had to stare at it for a second to see that. I'll fix that up.

Reply:
Fixed